fe/fe-core/src/main/java/com/starrocks/datacache/DataCacheSelectMetrics.java

```diff
@@ -0,0 +1,151 @@
+            .addColumn(new Column("WRITE_CACHE_TIME", ScalarType.createVarcharType()))
+            .addColumn(new Column("TOTAL_CACHE_USAGE", ScalarType.createVarcharType()))
+            .build();
+
+    private static final ShowResultSetMetaData VERBOSE_META_DATA = ShowResultSetMetaData.builder()
+            .addColumn(new Column("BE_IP", ScalarType.createVarcharType()))
+            .addColumn(new Column("STATUS", ScalarType.createVarcharType()))
+            .addColumn(new Column("ALREADY_CACHED_SIZE", ScalarType.createVarcharType()))
+            .addColumn(new Column("WRITE_CACHE_SIZE", ScalarType.createVarcharType()))
+            .addColumn(new Column("WRITE_CACHE_TIME", ScalarType.createVarcharType()))
+            .addColumn(new Column("TOTAL_CACHE_USAGE", ScalarType.createVarcharType()))
+            .build();
+
+    private final Map<Long, LoadDataCacheMetrics> beMetrics = new HashMap<>();
+
+    public void updateLoadDataCacheMetrics(long backendId, LoadDataCacheMetrics metrics) {
+        beMetrics.merge(backendId, metrics, LoadDataCacheMetrics::mergeMetrics);
+    }
+
+    public ShowResultSet getShowResultSet(boolean isVerbose) {
+        if (isVerbose) {
+            return getVerboseShowResultSet();
+        } else {
+            return getSimpleShowResultSet();
+        }
+    }
+
+    public Map<Long, LoadDataCacheMetrics> getBeMetrics() {
+        return beMetrics;
+    }
+
+    private ShowResultSet getVerboseShowResultSet() {
+        final SystemInfoService clusterInfoService = GlobalStateMgr.getCurrentState().getNodeMgr().getClusterInfo();
+        List<List<String>> rows = Lists.newArrayList();
+        for (Map.Entry<Long, LoadDataCacheMetrics> entry : beMetrics.entrySet()) {
+            List<String> row = Lists.newArrayList();
+            LoadDataCacheMetrics metrics = entry.getValue();
+
+            long backendId = entry.getKey();
+            Backend backend = clusterInfoService.getBackend(backendId);
+            if (backend == null) {
+                row.add("N/A");
+            } else {
+                row.add(backend.getIP());
+            }
+
+            row.add("SUCCESS");
+
+            row.add(metrics.getReadBytes().toString());
+            row.add(metrics.getWriteBytes().toString());
+            row.add(metrics.getWriteTimeNs().toString());
+            row.add(String.format("%.2f%%", metrics.getLastDataCacheMetrics().getCacheUsage() * 100));
+
+            rows.add(row);
+        }
+        return new ShowResultSet(VERBOSE_META_DATA, rows);
+    }
+
+    private ShowResultSet getSimpleShowResultSet() {
+        long alreadyCachedSize = 0;
+        long writeCacheSize = 0;
+        long writeCacheTime = 0;
+        long totalCacheSize = 0;
+        long totalUsedCacheSize = 0;
+
+        for (Map.Entry<Long, LoadDataCacheMetrics> entry : beMetrics.entrySet()) {
+            LoadDataCacheMetrics metrics = entry.getValue();
+            alreadyCachedSize += metrics.getReadBytes().getBytes();
```

Comment:
The `alreadyCachedSize` may be confusing if we calculate it by readbytes. For example, when we reading continuous 8M bytes which in one shared buffer, if the first megabyte data are not in datacache, we will read all 8M data from remote storage, while the last 7M may has been cached before.

Reply:
you are right, we need to fix it after we can calculate the accuracy hit rate